IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -436,7 +436,7 @@ - (void)acquireTokenWithRefreshTokenImpl:(MSIDBaseToken<MSIDRefreshableToken> *)
     MSIDRefreshTokenGrantRequest *tokenRequest = [self.oauthFactory refreshTokenRequestWithRequestParameters:self.requestParameters
                                                                                                 refreshToken:refreshToken.refreshToken];
     // Currently SilentTokenRequest has 3 child classes: Legacy, Default (local) and SSO. We will init the throttling service in Default and SSO and exclude Legacy. So the nil check of throttling service is needed
-    if (!self.throttlingService)
+    if (!self.throttlingService || [MSIDThrottlingService isThrottlingDisable])
```

Comment:
Is it possible to not pass ```throttlingService``` into this class so that we can avoid calling ```[MSIDThrottlingService isThrottlingDisable]``` couple times.

Basically to move more ```[MSIDThrottlingService isThrottlingDisable]``` to a upper layer

Reply:
Agree with Kai's suggestion

Reply:
Are you saying we want to add the check `[MSIDThrottlingService isThrottlingDisable]` in the initializer of `MSIDThrottlingService.h` or in the initializer of `MSIDDefaultSilentTokenRequest.h`? 
At first, I was thinking the same, but on the second thought, I'm not sure if coupling like that a good practice: for example if I want to change `MSIDThrottlingService isThrottlingDisable` in the future I need to consider changing the initializer, too.

Reply:
Maybe do the check before call this method like passing nil into it :)

This is a nit and could be easier for future maintaining, and it is update to your call